File: 66.加一.go
package main

/*
 * @lc app=leetcode.cn id=66 lang=golang
 *
 * [66] 加一
 */

// @lc code=start
func plusOne(digits []int) []int {
	n := len(digits)
	for i := n - 1; i >= 0; i-- {
		digits[i]++
		digits[i] %= 10
		if digits[i] != 0 {
			return digits
		}
	}
	digits = make([]int, n+1)
	digits[0] = 1
	return digits
}

// @lc code=end

File: 66.加一.js
/*
 * @lc app=leetcode.cn id=66 lang=javascript
 *
 * [66] 加一
 */

// @lc code=start
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
  const len = digits.length

  for (let i = len - 1; i >= 0; i--) {
    digits[i] += 1
    digits[i] %= 10
    if (digits[i] !== 0) return digits
  }

  const res = Array(len+1).fill(0)
  res[0] = 1
  return res
};
// @lc code=end

